### PR TITLE
Make RealmScans a working source (heavy maintenance)

### PR DESCRIPTION
**Pull request by Ivanmatthew**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,5 +6,14 @@
     "editor.formatOnSave": true, // optional 
     "editor.formatOnSaveMode": "file", // required to format on save
     "files.autoSave": "onFocusChange", // optional but recommended
-    "vs-code-prettier-eslint.prettierLast": false // set as "true" to run 'prettier' last not first
```
Comment (TheNetsky) [inline .vscode/settings.json:9]: ?

File: .gitignore [hunk 2]
```diff
@@ -123,3 +123,4 @@ docs
 .DS_Store
 todo
 package-lock.json
+/package-lock.json
```
Comment (TheNetsky) [inline .gitignore]: Why?

File: src/MangaStream.ts [hunk 3]
```diff
@@ -141,6 +141,11 @@ export abstract class MangaStream implements ChapterProviding, HomePageSectionsP
      */
     directoryPath = 'manga'
 
+    /**
+     * The pathname between the domain and the filter page (this is usually the same as the directory path).
+     */
+    filterPath = this.directoryPath
```
Comment (TheNetsky) [inline src/MangaStream.ts]: What's the purpose for this and why should it be added? Has not been needed before?

File: src/MangaStream.ts [hunk 4]
```diff
@@ -345,7 +350,7 @@ export abstract class MangaStream implements ChapterProviding, HomePageSectionsP
 
     async getSearchTags(): Promise<TagSection[]> {
         const request = App.createRequest({
-            url: `${this.baseUrl}/${this.directoryPath}/`,
+            url: `${this.baseUrl}/${this.filterPath}/`,
```
Comment (TheNetsky) [inline src/MangaStream.ts]: What's the purpose for this and why should it be added? Has not been needed before?

File: src/MangaStream.ts [hunk 5]
```diff
@@ -440,7 +445,7 @@ export abstract class MangaStream implements ChapterProviding, HomePageSectionsP
 
             // eslint-disable-next-line no-async-promise-executor
             promises.push(new Promise(async () => {
-                section.section.items = await this.parser.parseHomeSection($, section, this)
+                section.section.items = await this.parser.parseHomeSection($, section, this, this.baseUrl)
```
Comment (TheNetsky) [inline src/MangaStream.ts]: What's the purpose for this and why should it be added? Has not been needed before?

File: src/MangaStreamParser.ts [hunk 6]
```diff
@@ -28,9 +27,9 @@ export class MangaStreamParser {
...
         }
 
-        const author = $(`span:contains(${source.manga_selector_author}), .fmed b:contains(${source.manga_selector_author})+span, .imptdt:contains(${source.manga_selector_author}) i`).contents().remove().last().text().trim() // Language dependant
-        const artist = $(`span:contains(${source.manga_selector_artist}), .fmed b:contains(${source.manga_selector_artist})+span, .imptdt:contains(${source.manga_selector_artist}) i`).contents().remove().last().text().trim() // Language dependant
-        const image = this.getImageSrc($('img', 'div[itemprop="image"]'))
+        const author = $(`span:contains(${source.manga_selector_author}), .fmed b:contains(${source.manga_selector_author})+span, .imptdt:contains(${source.manga_selector_author}) i, .tsinfo > div:nth-child(4) > i`).contents().remove().last().text().trim() // Language dependant
+        const artist = $(`span:contains(${source.manga_selector_artist}), .fmed b:contains(${source.manga_selector_artist})+span, .imptdt:contains(${source.manga_selector_artist}) i, .tsinfo > div:nth-child(5) > i`).contents().remove().last().text().trim() // Language dependant
```
Comment (TheNetsky) [inline src/MangaStreamParser.ts]: For what purpose is this new selector? What source did not work with the previous ones?

File: src/MangaStreamParser.ts [hunk 7]
```diff
@@ -28,9 +27,9 @@ export class MangaStreamParser {
...
+        const image = this.getImageSrc($('img', 'div[itemprop="image"]'), undefined)
```
Comment (TheNetsky) [inline src/MangaStreamParser.ts]: What's the purpose for this and why should it be added? Has not been needed before?

File: src/MangaStreamParser.ts [hunk 8]
```diff
@@ -197,7 +196,7 @@ export class MangaStreamParser {
             }
 
             const title: string = $('a', obj).attr('title') ?? ''
-            const image = this.getImageSrc($('img', obj)) ?? ''
+            const image = this.getImageSrc($('img', obj), undefined) ?? ''
```
Comment (TheNetsky) [inline src/MangaStreamParser.ts]: What's the purpose for this and why should it be added? Has not been needed before?

File: src/MangaStreamParser.ts [hunk 9]
```diff
@@ -253,7 +252,7 @@ export class MangaStreamParser {
         for (const manga of mangas.toArray()) {
             const title = section.titleSelectorFunc($, manga)
 
-            const image = this.getImageSrc($('img', manga)) ?? ''
+            const image = this.getImageSrc($('img', manga), baseUrl) ?? ''
```
Comment (TheNetsky) [inline src/MangaStreamParser.ts]: What's the purpose for this and why should it be added? Has not been needed before?

File: src/MangaStreamParser.ts [hunk 10]
```diff
@@ -217,7 +216,7 @@ export class MangaStreamParser {
 
         for (const manga of $('div.bs', 'div.listupd').toArray()) {
             const title = $('a', manga).attr('title')
-            const image = this.getImageSrc($('img', manga)) ?? ''
+            const image = this.getImageSrc($('img', manga), undefined) ?? ''
```
Comment (TheNetsky) [inline src/MangaStreamParser.ts]: What's the purpose for this and why should it be added? Has not been needed before?

File: src/MangaStreamParser.ts [hunk 11]
```diff
@@ -317,9 +316,17 @@ export class MangaStreamParser {
         }
 
         image = image?.split('?resize')[0] ?? ''
-        image = image.replace(/^\/\//, 'https://')
-        image = image.replace(/^\//, 'https:/')
 
+        if (!image?.startsWith('http')) {
```
Comment (TheNetsky) [inline src/MangaStreamParser.ts]: Why is this changed? Pretty sure it worked fine before hand, if not give some context on which item it didn't work for.

File: .gitignore (hunk 2 above)
Comment (Ivanmatthew) [reply .gitignore]: My bad, it was intended to gitkeep package-lock, I'll add a patch commit to fix this.

File: .vscode/settings.json (hunk 1 above)
Comment (Ivanmatthew) [reply .vscode/settings.json:9]: This line of code is the exact same as from before

File: src/MangaStream.ts (hunk 3 above)
Comment (Ivanmatthew) [reply src/MangaStream.ts]: I'm trying to seperate values to make it future-proof. In hindsight, I don't think it'll be useful anymore and will remove it in a subsequent patch commit.

File: src/MangaStream.ts (hunk 4 above)
Comment (Ivanmatthew) [reply src/MangaStream.ts]: > I'm trying to seperate values to make it future-proof. In hindsight, I don't think it'll be useful anymore and will remove it in a subsequent patch commit.

File: src/MangaStream.ts (hunk 5 above)
Comment (Ivanmatthew) [reply src/MangaStream.ts]: Some mangastream sites use src attributes like "https://google.com/images/abc.png" and others '/images/abc.png".
I thought it might be a nice feature to have an already implemented system might we add some of these mangastream sources in the future.

File: src/MangaStreamParser.ts (hunk 6 above)
Comment (Ivanmatthew) [reply src/MangaStreamParser.ts]: It was put in the wrong file, my bad

File: src/MangaStreamParser.ts (hunk 7 above)
Comment (Ivanmatthew) [reply src/MangaStreamParser.ts]: > Some mangastream sites use src attributes like "https://google.com/images/abc.png" and others '/images/abc.png".
I thought it might be a nice feature to have an already implemented system might we add some of these mangastream sources in the future.

File: src/MangaStreamParser.ts (hunk 8 above)
Comment (Ivanmatthew) [reply src/MangaStreamParser.ts]: > Some mangastream sites use src attributes like "https://google.com/images/abc.png" and others '/images/abc.png".
I thought it might be a nice feature to have an already implemented system might we add some of these mangastream sources in the future.

File: src/MangaStreamParser.ts (hunk 10 above)
Comment (Ivanmatthew) [reply src/MangaStreamParser.ts]: > Some mangastream sites use src attributes like "https://google.com/images/abc.png" and others '/images/abc.png".
I thought it might be a nice feature to have an already implemented system might we add some of these mangastream sources in the future.

File: src/MangaStreamParser.ts (hunk 9 above)
Comment (Ivanmatthew) [reply src/MangaStreamParser.ts]: > Some mangastream sites use src attributes like "https://google.com/images/abc.png" and others '/images/abc.png".
I thought it might be a nice feature to have an already implemented system might we add some of these mangastream sources in the future.

**Comment by Ivanmatthew** [comment]
Squashed @TheNetsky

**Comment by Ivanmatthew** [comment]
We have come to the agreement that this does not belong in a generics-focused repository, as it has deviated way too much to still be considered "generic".
